server/server.go

```diff
@@ -416,13 +416,17 @@ func filterpath(peer *Peer, path, old *table.Path) *table.Path {
 		}
 
 		if ignore {
-			if !path.IsWithdraw && old != nil && old.GetSource().Address.String() != peer.ID() && old.GetSource().AS != peer.fsm.pConf.State.PeerAs {
-				// we advertise a route from ebgp,
-				// which is the old best. We got the
-				// new best from ibgp. We don't
-				// advertise the new best and need to
-				// withdraw the old.
-				return old.Clone(true)
+			if !path.IsWithdraw && old != nil {
+				oldSource := old.GetSource()
+				if oldSource.Address == nil || oldSource.Address.String() != peer.ID() && oldSource.AS != peer.AS() {
+					// We withdraw the old best which was injected locally
+					// (from CLI or gRPC for example) in order to avoid the
+					// old best left on peers.
```

Comment:
This happens only if this peer has the same prefix routes from an ibgp peer. Can you improve the comment?

Reply:
Sure, I will add some descriptions for it.